web/app.js

```diff
@@ -1739,14 +1741,41 @@ function webViewerPageRendered(evt) {
     });
     // It is a good time to report stream and font types.
     PDFViewerApplication.pdfDocument.getStats().then(function (stats) {
+      // Convert the stats to telemetry friendly names.
+      let newStats = {
+        streamTypes: toTelemtryNames(swapKeyValue(StreamType),
+                                     stats.streamTypes),
+        fontTypes: toTelemtryNames(swapKeyValue(FontType), stats.fontTypes),
+      };
+
       PDFViewerApplication.externalServices.reportTelemetry({
         type: 'documentStats',
-        stats,
+        stats: newStats,
       });
     });
   }
 }
 
+function toTelemtryNames(lookup, stats) {
+  // Replace the integer types with the string name from the constant.
+  let ret = [];
+  for (let key in stats) {
+    if (!stats[key]) {
+      continue;
+    }
+    ret.push(lookup[key].padEnd(3, 'X'));
+  }
+  return ret;
+}
+
+function swapKeyValue(obj) {
+  let ret = {};
```

Comment:
This can be `const`.

Reply:
Also, isn't the sort-of default in PDF.js to use `Object.create(null);` rather than `{}`, to avoid possible issues if somehow someone would mess with the Object prototype?